jscomp/test/record_regression.res

```diff
@@ -100,3 +100,47 @@ let setAA = (ao: option<int>) => {aa: @ns.optional ao, bb: None}
 //     @ns.optional bb: int,
 //   }
 // }
+
+type foo = Foo({name: string, age?: int})
+
+let foo1 = Foo({name: "foo"}) 
+let foo2 = Foo({name: "foo", age: 3})
+
+// should be type error
```

Comment:
I guess this is not needed.

Reply:
Removed https://github.com/rescript-lang/rescript-compiler/pull/5827/commits/651ba2b922c9a1406fa4eb985cffc09fd05e98c7